cornflow/schemas/schema_manager.py

```diff
@@ -0,0 +1,368 @@
+"""
+Class to help create and manage data schema and to validate json files.
+"""
+
+
+import json
+import re
+from jsonschema import validate
+from copy import deepcopy
+from genson import SchemaBuilder
+from ..schemas.schema_dict_functions import gen_schema, ParameterSchema, sort_dict
+from .constants import JSON_TYPES
+
+
+class SchemaManager:
+    
+    def __init__(self, schema_path=None):
+        """
+        Class to help create and manage data schema.
+        Once a schema is loaded, allow the validation of data.
+        """
+    
+        self.jsonschema = {}
+        self.types = JSON_TYPES
+        self.schema_dict = {"DataSchema": {}}
+        self.flask_schemas = {}
+        self.draft_schema = {}
+        if schema_path is not None:
+            self.load_schema(schema_path)
+    
+    def get_jsonschema(self):
+        """
+        Return a copy of the stored jsonschema.
+        """
+        return deepcopy(self.jsonschema)
+    
+    def is_schema_loaded(self):
+        """
+        Check if a schema is loaded.
+        """
+        loaded = len(self.jsonschema) > 0
+        if not loaded:
+            raise Warning("No jsonschema has been loaded in the SchemaManager class")
+        return loaded
+    
+    def validate_data(self, data_path):
+        """
+        Validate a json file according to the loaded jsonschema.
+
+        :param data_path The path to the json file containing the data.
+
+        :return: True if the data is valid and False if it is not.
+        """
+        self.is_schema_loaded()
+        
+        data = self.load_json(data_path)
+        try:
+            validate(data, self.get_jsonschema())
+            return True
+        except Exception as e:
+            print("json file not valid: ", e)
+            return False
+    
+    def jsonschema_to_dict(self):
+        """
+        Transform a jsonschema into a dictionary and store it in self.schema_dict
+        
+        :return: The schema dictionary
+        """
+        jsonschema = self.get_jsonschema()
+        
+        if not self.is_schema_loaded():
+            return {"DataSchema": {}}
```

Comment:
When is empty, couldn't you just return `self.schema_dict` ? Why is DataSchema is a dictionary here and later a list?

Reply:
done